crates/sui-types/src/messages.rs

```diff
@@ -437,6 +444,37 @@ where
         &self.gas_payment
     }
 
+    pub fn move_calls(&self) -> SuiResult<Vec<&MoveCall>> {
+        let move_calls = match &self.kind {
+            TransactionKind::Single(s) => {
+                let mut result = vec![];
+                let sub = s.move_call();
+                if sub.is_some() {
+                    result.push(sub.unwrap());
+                }
+                result
+            }
+            TransactionKind::Batch(b) => {
+                let mut result = vec![];
+                for kind in b {
+                    fp_ensure!(
+                        !matches!(kind, &SingleTransactionKind::Publish(..)),
+                        SuiError::InvalidBatchTransaction {
+                            error: "Publish transaction is not allowed in Batch Transaction"
+                                .to_owned(),
+                        }
+                    );
+                    let sub = kind.move_call();
+                    if sub.is_some() {
+                        result.push(sub.unwrap());
+                    }
```

Comment:
I would suggest
```
if let Some(call) = kind.move_call() {
     result.push(call)
}
```

or even more succinctly:

```
result.extend(kind.move_call().iter())
```

Reply:
(reason: avoid unwrap() whenever possible, as future readers of the code will have to stop and ask "can this unwrap() panic?")